echopop/computation/biology.py

```diff
@@ -104,83 +128,200 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
-         ]
+    ### Process biological datasets
+    # ---- Process haul data ( Station 2 - unsexed - unaged )
+    haul_strata_weight = (
+        haul_data
+        .groupby( 'stratum_num' )[ 'haul_weight' ]
+        .sum( )
+        .to_frame( 'stratum_weight' )
+        .reset_index( )
+        .assign( group = 'unaged' )
+    ) 
+
+    # ---- Process specimen data ( Station 1 - sexed - aged )
+    specimen_strata_weight = (
+        specimen_data
+        .groupby( 'stratum_num' )[ 'weight' ]
+        .sum( )
+        .to_frame( 'stratum_weight' )
+        .reset_index( )
+        .assign( group = 'aged' )
     )
 
+    ### Merge the two dataframes
+    # Yields the summed weights per stratum and station (or aged/unaged designation)
+    weight_strata_aged_unaged = pd.concat( [ haul_strata_weight , specimen_strata_weight ] )
+ 
     ### Sum weights for each stratum
     weight_strata = (
-            weight_strata_station
-            .groupby( [ 'stratum_num' ] )
-            .apply( lambda x: pd.Series( { 'weight_stratum_total': x.stratum_weight.sum( ) } ) )            
-            .reset_index( )
-        )
+        weight_strata_aged_unaged
+        .groupby( "stratum_num" )[ "stratum_weight" ]
+        .sum( )
+        .to_frame( 'weight_stratum_all' )
+        .reset_index( )
+    )
     
     ### Carriage return
-    return weight_strata_station , weight_strata
+    return weight_strata_aged_unaged , weight_strata
 
-def sum_strata_length_age_sex_weight( haul_data: pd.DataFrame ,
-                                      specimen_data: pd.DataFrame ,
-                                      length_intervals: np.ndarray ,
-                                      age_intervals: np.ndarray ):
-    
-    ### Sum weights from each station and strata 
-    weight_strata_station , weight_strata = sum_strata_weight( haul_data ,
-                                                               specimen_data )
-    
-    ### Calculate summed length-age-sex weight bins across strata
-    weight_length_age_sex_stratum = (
+def compute_index_aged_weight_proportions( specimen_data: pd.DataFrame ,
+                                           length_intervals: np.ndarray ,
+                                           age_intervals: np.ndarray ):
+    """
+    Calculate length-age binned weight proportions    
+
+    Parameters
+    ----------
+    specimen_data: pd.DataFrame
+        Dataframe containing specimen weights
+    length_intervals: np.ndarray
+        Array containing length bins/intervals
+    age_intervals: np.ndarray
+        Array containing age bins/intervals
+    """ 
+
+    ### Process the specimen data 
+    specimen_binned = (
         specimen_data
+        # ---- Drop unaged fish 
+        # !!! TODO: pending what FEAT says, weights associated with 
+        # missing ages should be added into the 'unaged' category. 
+        # This would further mean apportioning these into `weight_strata_aged_uanged`
         .dropna( how = 'any' )
```

Comment:
May be a good idea to specify the column to avoid accidental drops
```suggestion
        .dropna( how = 'any' , subset=AGE_COL)
```